src/components/swap/components/transaction-review/index.tsx

```diff
@@ -190,15 +167,46 @@ const ContractSection = ({
   )
 }
 
+function useBestQuote(review: TransactionReview) {
+  const { quoteResult } = review
+  const bestQuote = useMemo(() => {
+    const isFlashmintBestQuote = quoteResult.bestQuote === QuoteType.flashmint
+    return isFlashmintBestQuote
+      ? quoteResult.quotes.flashmint
+      : quoteResult.quotes.zeroex
+  }, [quoteResult])
+  return { bestQuote }
+}
+
+const useTradeMaker = (quote: Quote | null) => {
+  const { executeTrade, isTransacting, txWouldFail } = useTrade()
+
+  /**
+   * Returns boolean indicating success or null for config error
+   */
+  const makeTrade = async (override: boolean) => {
+    if (!quote) return null
+    try {
+      await executeTrade(quote, override)
+      return true
+    } catch {
+      return false
```

Comment:
Make sense to add any logging here, or is it automatically displayed if something fails?

Reply:
I'll add logging for this with upcoming task of adding a new error logging provider (if not logged automatically).